### PR TITLE
PR localization of views - logix

### PR DESCRIPTION
**Pull request by pavanjoshi914**
#### Describe the changes you have made in this PR -
localize logix module
extract repeated keys present in about page module and logix module in general namespace

### Screenshots of the changes (If any) -
image

image

image

image

image

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

**Review by ShreyaPrasad1209 — APPROVED** [review]

**Review by Shivansh2407 — CHANGES_REQUESTED** [review]
Great Work @pavanjoshi914 :tada: I have suggested a few minor changes, please update them and we'll be good to go.

File: config/locales/views/logix/hi.yml [hunk 1]
```diff
@@ -1,14 +1,151 @@
...
+      title: "सर्किटवर्स - ऑनलाइन डिजिटल लॉजिक सर्किट सिम्युलेटर"
+      main_heading: "लॉजिक सर्किट की दुनिया में मुफ्त में गोता लगाएँ!"
+      main_description: "साधारण गेट से लेकर काम्प्लेक्स सेकेंटिअल सर्किट्स, प्लॉट टाइमिंग डायग्राम, आटोमेटिक सर्किट जनरेशन, स्टैण्डर्ड आईसी के बारे में अधिक, और बहुत कुछ जानिए"
+      learn_logic_design: "लॉजिक डिज़ाइन सीखें"
+      for_teachers: "शिक्षकों के लिए"
+      for_contributors: "योगदानकर्ताओं के लिए"
+      explore_more: "इसे और अधिक एक्सप्लोर करें"
+      features:
+        main_heading: "विशेषताएं"
+        main_description: "ड्रैग-एंड-ड्रॉप, कॉपी/पेस्ट, ज़ूम, और अधिक विशेषताओं का उपयोग करके आधुनिक और सहज यूजर इंटरफ़ेस के साथ डिजिटल सर्किट की रचना जल्दी और आसानी से करे।"
```
[the remark this answers is not in the document]
Comment (pavanjoshi914) [reply config/locales/views/logix/hi.yml:43]: image

its rendering on my local system

[the remark this answers is not in the document]
Comment (Shivansh2407) [reply config/locales/views/logix/hi.yml:43]: It should be ड्रैग-एंड-ड्रॉप right? But it becomes different in the screenshot

[the remark this answers is not in the document]
Comment (pavanjoshi914) [reply config/locales/views/logix/hi.yml:43]: can you share me screenshot how it is rendered on your side

File: config/locales/views/logix/hi.yml [hunk 2]
```diff
@@ -1,14 +1,151 @@
...
+      become_patreon: "एक पैट्रियन बनें"
+      donate_via_paypal: "पेपैल के माध्यम से दान करें"
+    examples:
+      title: "सर्किटवर्स - सर्किट के उदाहरण"
+      main_heading: "सर्किट के उदाहरण"
+      main_description: "सर्वोत्तम उदाहरणों और उपयोगकर्ताओं से सीखें और प्रेरित होइए"
+    index:
+      title: "सर्किटवर्स - ऑनलाइन डिजिटल लॉजिक सर्किट सिम्युलेटर"
+      main_heading: "लॉजिक सर्किट की दुनिया में मुफ्त में गोता लगाएँ!"
+      main_description: "साधारण गेट से लेकर काम्प्लेक्स सेकेंटिअल सर्किट्स, प्लॉट टाइमिंग डायग्राम, आटोमेटिक सर्किट जनरेशन, स्टैण्डर्ड आईसी के बारे में अधिक, और बहुत कुछ जानिए"
```
[the remark this answers is not in the document]
Comment (pavanjoshi914) [reply config/locales/views/logix/hi.yml]: typo corrected

File: config/locales/views/logix/hi.yml [hunk 3]
```diff
@@ -1,14 +1,151 @@
...
+      editor_picks:
+        main_heading: "संपादक की पसंद"
+        main_description: "इन सर्किटों को हमारे लेखकों ने उनकी उत्कृष्टता के लिए चुना है"
+      recent_circuits:
+        main_heading: "हाल के सर्किट"
+        main_description: "हमारे बेहतर उपयोगकर्ताओं द्वारा नवीनतम सर्किट के बारे में और जाने"
+    teachers:
+      title: "सर्किटवर्स के शिक्षक"
+      main_heading: "शिक्षकों के लिए"
+      main_description: "सर्किटविर्स को कक्षा में उपयोग करने के लिए बहुत आसान बनाया गया है। मंच में शिक्षकों को कक्षा और असाइनमेंट में सहायता करने के लिए सुविधाएँ हैं।"
```
[the remark this answers is not in the document]
Comment (pavanjoshi914) [reply config/locales/views/logix/hi.yml]: checked all the words starting from "circuit" typo is only at one place , corrected!!

File: config/locales/views/logix/hi.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Shivansh2407) [reply config/locales/views/logix/hi.yml:43]: > image
> 
> its rendering on my local system

You can notice some difference in the above screenshot that you've shared, and in this word:ड्रैग-एंड-ड्रॉप

 SS -> ड्रैग-एंड-ड्रॉप

**Comment by tachyons** [comment]
logix was the original name of circuitverse, not need to refer that anymore. also, 'general', Instead of `general.xyz`, you van just use `xyz`

**Comment by pavanjoshi914** [comment]
I am using rule  that translations will be put in ```folder_name + view_name````  namespace. as this will bring scalability and reduce key conflicts a lot in future.

here 
```
logix:
  contribute:
    ....
  ```
shall I change toplevel namespace from ```logix:``` to ```circuitverse:``` ?
this will break consistency as folder name is still logix

regarding ```general``` I used this word to define context  that is key is repeating key and used in other templates also.

if we just use ```email_us_heading```  contributors will not get idea that this key is also used in other templates also this key is placed in general folder.

Definitely it will reduce one level of nesting. To what I shall give priority here

File: config/locales/views/logix/hi.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Shivansh2407) [reply config/locales/views/logix/hi.yml:43]: As discussed on slack, this is some font rendering issue which can be skipped for now.

**Review by Shivansh2407 — APPROVED** [review]
Great Work @pavanjoshi914 :tada:

File: config/locales/views/logix/hi.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (pavanjoshi914) [reply config/locales/views/logix/hi.yml:43]: already discussed with @tachyons

**Comment by tachyons** [comment]
For the general use config/locales/locale.yml, like 'config/locales/hi.yml'

**Comment by pavanjoshi914** [comment]
changes made accordingly !!
CC @tachyons

**Pull request merged**